### PR TITLE
fix: Improve text representation for boolean values in ChoicesFormatter

### PR DESCRIPTION
**Pull request by fit2bot**
fix: Improve text representation for boolean values in ChoicesFormatter

### CODE REVIEW

File: src/components/Table/TableFormatters/ChoicesFormatter.vue [hunk 1]
```diff
@@ -24,16 +24,17 @@ const formatterArgsDefault = {
...
-    return (cellValue && typeof cellValue === 'object') ? cellValue.label : this.textChoices[key] || cellValue
+    if (cellValue && typeof cellValue === 'object') {
+      return cellValue.label
+    }
+    if (key === true || key === 'true') return i18n.t('Yes')
+    if (key === false || key === 'false') return i18n.t('No')
+    return cellValue
   },
   getIcon({ row, cellValue }) {
     const key = this.getKey({ row, cellValue })
```
Comment (baijiangjie) [inline src/components/Table/TableFormatters/ChoicesFormatter.vue:40]: The code includes some redundant lines of formatting, which can be simplified to enhance readability and maintainability. The use of object keys in `getKey()`, although helpful in certain contexts, could be removed as it does not strictly increase efficiency over using an index directly, but would still leave a few characters shorter than the alternative.

Additionally, the current implementation uses string constants (`i18n`), but considering modern JavaScript practices and best coding practices like ESLint's usage for linting strings within code elements or better using variables that reference the proper constant rather than hardcoding them, this may also lead to further benefits.

To fully optimize and streamline the code without impacting functionality, additional unit tests should be added to ensure all functionalities work as intended across various scenarios, including edge cases, especially with regards to handling non-string values, data types other than boolean or text, arrays, etc.

For example:
- Consider wrapping complex operations such as array indexing into functions where possible, making the code more flexible yet simpler.
- Ensure consistency in variable naming and logic flow.
- Use modern programming practices such as TypeScript/JavaScript ES modules instead of local files containing static HTML content to reduce dependency cycle, improve organization and facilitate future updates.

Here is how we might apply this approach:

```
// In your project's root file:
import I18n from './utils/I18n.js'; // Assuming this import provides the required constant here

class DataFormatter {
  formatterArgsDefault = {
    true: 'text-primary',
    false: 'text-danger'
  };

  textChoices = {
    true: I18n.t('Yes'),
    false: I18n.t('No')
  };

  getKey(row) {
    return row.getValue();
  }

  getText(cellValue) {
    const key = this.getKey({ row: cellValue });

    if (typeof key === 'string' && /[\s\t\r\n]+/.test(key)) { 
      const lowerCaseKey = key.toLowerCase();
      
      switch(lowerCaseKey){
        case 'yes':
          return I18n.t('Yes');
        case 'no':
          return I18n.t('No'); 
      default:
        return null;
      }
    }

    // If neither "true" nor "false", assume it's just the cell value itself
    return cellValue;

```
